cx/dependency_guardian/guardian.py

```
@@ -0,0 +1,106 @@
+
+        for line in output:
+            if line.startswith("  Depends:"):
+                pkg = line.split("Depends:")[1].strip()
+                incoming_deps.append(pkg.split(' ')[0].replace('<', '').replace('>', ''))
+            elif line.startswith("  Conflicts:"):
+                pkg = line.split("Conflicts:")[1].strip()
+                incoming_conflicts.append(pkg.split(' ')[0].replace('<', '').replace('>', ''))
+
+        return self._analyze_conflicts(target_package, incoming_deps, incoming_conflicts)
+
+    def _analyze_conflicts(self, target: str, incoming_deps: List[str], incoming_conflicts: List[str]) -> ConflictReport:
```

Ruff (0.15.2)

[warning] Unused method argument: `incoming_deps`

(ARG002)